src/overlays/actors/ovl_Boss_Va/z_boss_va.c

```diff
@@ -173,157 +294,3728 @@ static ColliderQuadInit D_8095BFF4 = {
+                        effect->rot.x = -0x4000;
+                    }
+
+                    if (!effect->timer) {
+                        effect->type = VA_NONE;
+                    }
+
+                } else if (effect->mode == GORE_FADING) {
+                    if (effect->timer == 0) {
+                        effect->type = VA_NONE;
+                        if (1) {}
+                    }
+
+                } else {
+                    Math_SmoothStepToF(&effect->scaleMod, 0.075f, 1.0f, 0.005f, 0.0f);
+                    Math_SmoothStepToF(&effect->vaGorePulseRate, 0.0f, 0.6f, 0.005f, 0.0013f);
+                    if ((globalCtx->gameplayFrames % 4) == 0) {
+                        Math_SmoothStepToS(&effect->primColor[0], 95, 1, 1, 0);
+                    }
+                }
+                effect->vaGorePulse += effect->vaGorePulseRate;
+            }
+
+            if (effect->type == VA_TUMOR) {
+                refActor = effect->parent;
+
+                effect->rot.z += 0x157C;
+                effect->envColor[3] = (s16)(Math_SinS(effect->rot.z) * 50.0f) + 80;
+                Math_SmoothStepToF(&effect->scale, effect->scaleMod, 1.0f, 0.01f, 0.005f);
+                effect->pos.x = effect->offset.x + refActor->actor.world.pos.x;
+                effect->pos.y = effect->offset.y + refActor->actor.world.pos.y;
+                effect->pos.z = effect->offset.z + refActor->actor.world.pos.z;
+
+                switch (effect->mode) {
+                    case TUMOR_UNUSED:
+                        if (effect->timer == 0) {
+                            yaw = Math_Vec3f_Yaw(&refActor->actor.world.pos, &effect->pos);
+                            effect->type = VA_NONE;
+                            BossVa_BloodSplatter(globalCtx, effect, yaw, effect->scale * 4500.0f, 1);
+                            BossVa_Gore(globalCtx, effect, yaw, effect->scale * 1.2f);
+                        }
+                        break;
+                    case TUMOR_BODY:
+                    case TUMOR_ARM:
+                        if (refActor->burst) {
+                            effect->type = VA_NONE;
+                            yaw = Math_Vec3f_Yaw(&refActor->actor.world.pos, &effect->pos);
+                            BossVa_BloodSplatter(globalCtx, effect, yaw, effect->scale * 4500.0f, 1);
+                            BossVa_Gore(globalCtx, effect, yaw, effect->scale * 1.2f);
+                        }
+                        break;
+                }
+                if (1) {}
+            }
+        }
+    }
+}
+
+void BossVa_DrawEffects(BossVaEffect* effect, GlobalContext* globalCtx) {
+    static void* sSparkBallTex[] = {
+        gBarinadeSparkBall1Tex, gBarinadeSparkBall2Tex, gBarinadeSparkBall3Tex, gBarinadeSparkBall4Tex,
+        gBarinadeSparkBall5Tex, gBarinadeSparkBall6Tex, gBarinadeSparkBall7Tex, gBarinadeSparkBall8Tex,
+    };
+    s16 i;
+    GraphicsContext* localGfx = globalCtx->state.gfxCtx;
```

Comment:
can rename this to gfxCtx. this was prob named before we had the disp macros use `__gfxCtx`